wolfcrypt/src/poly1305.c

```
@@ -88,15 +88,17 @@ static word32 cpu_flags_set = 0;
 #endif
 
 #if defined(USE_INTEL_POLY1305_SPEEDUP) || defined(POLY130564)
-    #if defined(_MSC_VER)
+    #if defined(__WATCOMC__)
+        #error "POLY130564 || USE_INTEL_POLY1305_SPEEDUP Watcom not supported"
```

Do we really need to throw an error here?

---

Yes, the alternative is a very unintuitive error.

Example config:

```
cmake -B build-watcom -G "Watcom WMake" ^
  -DCMAKE_C_FLAGS="-DWOLFSSL_DLL -DBUILDING_WOLFSSL -DPOLY130564 " ^
  -DCMAKE_SYSTEM_NAME=Windows ^
  -DCMAKE_EXECUTABLE_SUFFIX=.exe ^
  -DCMAKE_SYSTEM_PROCESSOR=i386 ^
  -DCMAKE_VERBOSE_MAKEFILE=TRUE ^
  -DWOLFSSL_ASM=NO ^
  -DWOLFSSL_EXAMPLES=YES ^
  -DWOLFSSL_CRYPT_TESTS=NO ^
  -DWOLFSSL_SINGLE_THREADED=YES ^
  -DBUILD_SHARED_LIBS=NO ^
  -DCMAKE_POLICY_DEFAULT_CMP0136=NEW ^
  -DCMAKE_WATCOM_RUNTIME_LIBRARY=SingleThreaded
if errorlevel 1 (
    echo "Config 2 failed!"
    exit /b
)
cmake --build build-watcom --target client
```

results in this error:

```
Open Watcom C/C++ x86 32-bit Compile and Link Utility
Version 2.0 beta Jul 11 2025 02:52:46 (32-bit)
Copyright (c) 2002-2025 The Open Watcom Contributors. All Rights Reserved.
Portions Copyright (c) 1988-2002 Sybase, Inc. All Rights Reserved.
Source code is available under the Sybase Open Watcom Public License.
See https://github.com/open-watcom/open-watcom-v2#readme for details.
        wcc386 C:\workspace\wolfssl-pr-cert\wolfcrypt\src\poly1305.c  -d+ -DBUILDING_WOLFSSL -DECC_SHAMIR -DECC_TIMING_RESISTANT -DGCM_TABLE_4BIT -DHAVE_AESGCM -DHAVE_CHACHA -DHAVE_CONFIG_H -DHAVE_DH_DEFAULT_PARAMS -DHAVE_ECC -DHAVE_ENCRYPT_THEN_MAC -DHAVE_EXTENDED_MASTER -DHAVE_FFDHE_2048 -DHAVE_HASHDRBG -DHAVE_HKDF -DHAVE_ONE_TIME_AUTH -DHAVE_POLY1305 -DHAVE_SUPPORTED_CURVES -DHAVE_THREAD_LS -DHAVE_TLS_EXTENSIONS -DNO_CHACHA_ASM -DNO_DES3 -DNO_DES3_TLS_SUITES -DNO_DSA -DNO_MD4 -DNO_MD5 -DNO_OLD_TLS -DNO_PSK -DNO_RC4 -DSINGLE_THREADED -DTFM_ECC256 -DTFM_NO_ASM -DTFM_TIMING_RESISTANT -DWC_NO_ASYNC_THREADING -DWC_RSA_BLINDING -DWC_RSA_PSS -DWOLFSSL_HAVE_MIN -DWOLFSSL_HAVE_MAX -DNO_WRITEV -DWOLFSSL_IGNORE_FILE_WARN -DWOLFSSL_NO_ASM -DWOLFSSL_NO_SHAKE128 -DWOLFSSL_NO_SHAKE256 -DWOLFSSL_PSS_LONG_SALT -DWOLFSSL_SHA384 -DWOLFSSL_SHA512 -DWOLFSSL_SYS_CA_CERTS -DWOLFSSL_TLS13 -DWOLFSSL_USE_ALIGN -I"C:\workspace\wolfssl-pr-cert" -I"C:\workspace\wolfssl-pr-cert\build-watcom" -I"C:\watcom\h" -I"C:\watcom\h\nt" -DWOLFSSL_DLL -DBUILDING_WOLFSSL -DPOLY130564 -wx -wcd=202 -d2 -foCMakeFiles\wolfssl.dir\wolfcrypt\src\poly1305.c.obj
Open Watcom C x86 32-bit Optimizing Compiler
Version 2.0 beta Jul 11 2025 02:49:30 (32-bit)
Copyright (c) 2002-2025 The Open Watcom Contributors. All Rights Reserved.
Portions Copyright (c) 1984-2002 Sybase, Inc. All Rights Reserved.
Source code is available under the Sybase Open Watcom Public License.
See https://github.com/open-watcom/open-watcom-v2#readme for details.
C:\workspace\wolfssl-pr-cert\wolfcrypt\src\poly1305.c(288): Error! E1011: Symbol 'word128' has not been declared
C:\workspace\wolfssl-pr-cert\wolfcrypt\src\poly1305.c(288): Error! E1009: Expecting ';' but found 'd0'
C:\workspace\wolfssl-pr-cert\wolfcrypt\src\poly1305.c(288): Error! E1011: Symbol 'd0' has not been declared
C:\workspace\wolfssl-pr-cert\wolfcrypt\src\poly1305.c(288): Error! E1011: Symbol 'd1' has not been declared
C:\workspace\wolfssl-pr-cert\wolfcrypt\src\poly1305.c(288): Error! E1011: Symbol 'd2' has not been declared
C:\workspace\wolfssl-pr-cert\wolfcrypt\src\poly1305.c(288): Error! E1011: Symbol 'd' has not been declared
C:\workspace\wolfssl-pr-cert\wolfcrypt\src\poly1305.c(313): Warning! W131: No prototype found for function 'MUL'
C:\workspace\wolfssl-pr-cert\wolfcrypt\src\poly1305.c(313): Warning! W131: No prototype found for function 'ADD'
C:\workspace\wolfssl-pr-cert\wolfcrypt\src\poly1305.c(318): Warning! W131: No prototype found for function 'SHR'
C:\workspace\wolfssl-pr-cert\wolfcrypt\src\poly1305.c(318): Warning! W131: No prototype found for function 'LO'
C:\workspace\wolfssl-pr-cert\wolfcrypt\src\poly1305.c(319): Warning! W131: No prototype found for function 'ADDLO'
C:\workspace\wolfssl-pr-cert\wolfcrypt\src\poly1305.c(313): Warning! W308: The function 'MUL' without prototyped parameters called
C:\workspace\wolfssl-pr-cert\wolfcrypt\src\poly1305.c(313): Note! N2002: 'MUL' defined in: C:\workspace\wolfssl-pr-cert\wolfcrypt\src\poly1305.c(313)

[  ... snip similar errors ... ]

C:\workspace\wolfssl-pr-cert\wolfcrypt\src\poly1305.c(320): Note! N2002: 'LO' defined in: C:\workspace\wolfssl-pr-cert\wolfcrypt\src\poly1305.c(318)
C:\workspace\wolfssl-pr-cert\wolfcrypt\src\poly1305.c: 1080 lines, included 153253, 28 warnings, 6 errors
Error: Unable to invoke "C:\watcom\binnt\wcc386.exe"
Error(E42): Last command making (CMakeFiles\wolfssl.dir\wolfcrypt\src\poly1305.c.obj) returned a bad status
Error(E02): Make execution terminated
Error(E42): Last command making (CMakeFiles\wolfssl.dir\all) returned a bad status
Error(E02): Make execution terminated
Error(E42): Last command making (CMakeFiles\client.dir\rule) returned a bad status
Error(E02): Make execution terminated
Error(E42): Last command making (client) returned a bad status
Error(E02): Make execution terminated
```

It might be resolvable, but I didn't chase it for this PR.